### PR TITLE
chore: Update error message for subnet methods that are not allowed through ingress messages

### PR DESCRIPTION
**Pull request by dsarlis**
The error message for subnet methods that were not allowed through ingress was coupled with the error message of a method not found which can be misleading when an ingress message is not inducted. Additionally, improve the error message for `SubnetMethodNotAllowed` when parsing an ingress to mention directly that the message is not allowed through ingress instead of "can only be called by canisters". The two are not exactly equivalent, the example here being `fetch_canister_logs` which can only be called by non-replicated queries.

### CODE REVIEW

File: rs/tests/src/execution/api_tests.rs [hunk 1]
```diff
@@ -379,7 +379,7 @@ pub fn node_metrics_history_ingress_update_fails(env: TestEnv) {
             assert_reject_msg(
                 result,
                 RejectCode::CanisterReject,
-                "ic00 method node_metrics_history can be called only by a canister",
+                "ic00 method node_metrics_history can be not be called via ingress messages",
```
Comment (basvandijk) [inline rs/tests/src/execution/api_tests.rs:382]:
```suggestion
                "ic00 method node_metrics_history can not be called via ingress messages",
```

File: rs/types/types/src/messages/ingress_messages.rs [hunk 2]
```diff
@@ -468,7 +468,7 @@ impl ParseIngressError {
             ParseIngressError::SubnetMethodNotAllowed => UserError::new(
                 ErrorCode::CanisterRejectedMessage,
                 format!(
-                    "ic00 method {} can be called only by a canister",
+                    "ic00 method {} can be not be called via ingress messages",
```
Comment (basvandijk) [inline rs/types/types/src/messages/ingress_messages.rs:471]:
```suggestion
                    "ic00 method {} can not be called via ingress messages",
```

**Review by basvandijk — APPROVED** [review]

**Review by ulan — APPROVED** [review]

**Comment by derlerd-dfinity** [comment]
Is there a potential forward compatibility issue? Say a version with this commit is running on mainnet and a rollback to a previous version without this commit happens, then we would fail deserializing the ingress history if it already contains a response with the new error type, no?

**Comment by dsarlis** [comment]
> Is there a potential forward compatibility issue? Say a version with this commit is running on mainnet and a rollback to a previous version without this commit happens, then we would fail deserializing the ingress history if it already contains a response with the new error type, no?

I'm not adding any new error type. `CanisterRejectedMessage` already exists. Are you thinking of something else?

**Comment by derlerd-dfinity** [comment]
Thanks for the clarification @dsarlis. I had missed that you are not introducing this new error type.

LGTM.

**Review by derlerd-dfinity — APPROVED** [review]

**Comment by derlerd-dfinity** [comment]
> Thanks for the clarification @dsarlis. I had missed that you are not introducing anything new that would be persisted.
> 
> LGTM.

Which brings up the related question whether we should move `IngressInductionError` out of ReplicatedState to make it obvious that it is not persisted (not in this MR)?

**Comment by dsarlis** [comment]
> > Thanks for the clarification @dsarlis. I had missed that you are not introducing anything new that would be persisted.
> > LGTM.
> 
> Which brings up the related question whether we should move `IngressInductionError` out of ReplicatedState to make it obvious that it is not persisted (not in this MR)?

Good point. I can take a look at this. It won't be an MR though :P.

**Pull request merged**